src/payment/api.rs

```diff
@@ -0,0 +1,478 @@
+    pub cancel_debit_note_timeout: Option<f64>,
+    pub send_invoice_timeout: Option<f64>,
+    pub cancel_invoice_timeout: Option<f64>,
+}
+
+impl ApiConfig {
+    pub fn from_env() -> envy::Result<Self> {
+        envy::from_env()
+    }
+}
+
+#[derive(Clone)]
+pub struct PaymentApi {
+    client: WebClient,
+    config: Arc<ApiConfig>,
+}
+
+impl WebInterface for PaymentApi {
+    const API_URL_ENV_VAR: &'static str = crate::payment::PAYMENT_URL_ENV_VAR;
+    const API_SUFFIX: &'static str = ya_client_model::payment::PAYMENT_API_PATH;
+
+    fn from_client(client: WebClient) -> Self {
+        let config = ApiConfig::default();
+        let config = Arc::new(config);
+        Self { client, config }
+    }
+}
+
+impl PaymentApi {
+    pub fn new(client: &WebClient, config: ApiConfig) -> Self {
+        let config = config.into();
+        let client = client.clone();
+        Self { client, config }
+    }
+
+    // accounts
+
+    pub async fn get_requestor_accounts(&self) -> Result<Vec<Account>> {
+        self.client.get("requestorAccounts").send().json().await
+    }
+
+    pub async fn get_provider_accounts(&self) -> Result<Vec<Account>> {
+        self.client.get("providerAccounts").send().json().await
+    }
+
+    // allocations
+
+    pub async fn create_allocation(&self, allocation: &NewAllocation) -> Result<Allocation> {
+        self.client
+            .post("allocations")
+            .send_json(allocation)
+            .json()
+            .await
+    }
+
+    pub async fn get_allocations<Tz>(
+        &self,
+        after_timestamp: Option<DateTime<Tz>>,
+        max_items: Option<u32>,
+    ) -> Result<Vec<Allocation>>
+    where
+        Tz: TimeZone,
+        Tz::Offset: Display,
+    {
+        #[allow(non_snake_case)]
+        let afterTimestamp = after_timestamp.map(|dt| dt.to_rfc3339());
+        #[allow(non_snake_case)]
+        let maxItems = max_items;
+
+        #[rustfmt::skip]
+        let url = url_format!(
+            "allocations",
+            #[query] afterTimestamp,
+            #[query] maxItems
+        );
+        self.client.get(&url).send().json().await
+    }
+
+    pub async fn get_allocation(&self, allocation_id: &str) -> Result<Allocation> {
+        let url = url_format!("allocations/{allocation_id}", allocation_id);
+        self.client.get(&url).send().json().await
+    }
+
+    pub async fn amend_allocation(&self, allocation: &Allocation) -> Result<Allocation> {
+        let allocation_id = &allocation.allocation_id;
+        let url = url_format!("allocations/{allocation_id}", allocation_id);
+        self.client.put(&url).send_json(allocation).json().await
+    }
+
+    pub async fn release_allocation(&self, allocation_id: &str) -> Result<()> {
+        let url = url_format!("allocations/{allocation_id}", allocation_id);
+        self.client.delete(&url).send().json().await
+    }
+
+    pub async fn decorate_demand(&self, allocation_ids: Vec<String>) -> Result<MarketDecoration> {
```

Comment:
It would be nice to have the function named in accordance with endpoint name.

```suggestion
    pub async fn get_demand_decorations(&self, allocation_ids: Vec<String>) -> Result<MarketDecoration> {
```

Reply:
done :)